### PR TITLE
Add coverage for kafka data recorder

### PR DESCRIPTION
**Pull request by zhouzhuojie**

### CODE REVIEW

File: pkg/handler/data_recorder_kafka.go [hunk 1]
```diff
@@ -143,7 +143,7 @@ func (r *kafkaEvalResult) Length() int {
 
 // Key generates the partition key
 func (r *kafkaEvalResult) Key() string {
-	if r.EvalResult == nil {
+	if r.EvalResult == nil || r.EvalContext == nil {
```
Comment (rstratton) [inline pkg/handler/data_recorder_kafka.go]: What does this change do?

Comment (zhouzhuojie) [reply pkg/handler/data_recorder_kafka.go]: an edge case found via the unit test. The line below `r.EvalContext.EntityID` requires that r.EvalContext is not nil
